### PR TITLE
[Claiming] - Claim page 2

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

This is cleaned up version of previous `claim-page` branch
This PR adds some functionality to the claim-page UI like

- uses the claim data from the hook and displays it
- adds functionality to the address/ens search input
- displays different content based on the claim data (no claims, airdrop only, investment claims)
- currently uses hardcoded claim data
- works only on Rinkeby for now
- for now uses only component state

### CODE REVIEW

File: src/custom/state/claim/hooks/utils.ts [hunk 1]
```diff
@@ -25,5 +34,100 @@ export function hasFreeClaim(claims: UserClaims | null): boolean {
...
+export function getFreeClaims(claims: UserClaims): UserClaims {
+  return claims?.filter((claim) => FREE_CLAIM_TYPES.includes(claim.type))
+}
+
+/**
+ * Helper function to transform claim data amount to CurrencyAmount
+ *
+ */
+export function parseClaimAmount(value: string, chainId: number | undefined): CurrencyAmount<Token> | undefined {
+  const vCow = chainId ? V_COW[chainId || 4] : undefined
```
Comment (W3stside) [inline src/custom/state/claim/hooks/utils.ts:63]: careful with this kind of code, don't forget to remove the 4 here

File: src/custom/state/claim/hooks/utils.ts [hunk 2]
```diff
@@ -25,5 +34,100 @@ export function hasFreeClaim(claims: UserClaims | null): boolean {
...
+
+  if (chainId === SupportedChainId.XDAI) {
+    map.UserOption = 'XDAI'
+  }
+
+  return map
+}
+
+export type TypeToPriceMapper = {
+  [key: string]: number
```
Comment (W3stside) [inline src/custom/state/claim/hooks/utils.ts:98]: better as `Record<string, number>` right?

File: src/custom/state/claim/hooks/utils.ts [hunk 3]
```diff
@@ -25,5 +34,100 @@ export function hasFreeClaim(claims: UserClaims | null): boolean {
...
+
+/**
+ * Helper function to get vCow price based on claim type and chainId
+ *
+ * @param type
+ */
+export function getTypeToPriceMap(): TypeToPriceMapper {
+  // Hardcoded values
+  const map: TypeToPriceMapper = {
+    GnoOption: 16.66,
```
Comment (W3stside) [inline src/custom/state/claim/hooks/utils.ts:109]: do we not have a type already for the keys here? an enum for example?

File: src/custom/state/claim/hooks/index.ts [hunk 4]
```diff
@@ -22,17 +22,27 @@ export * from './hooksMod'
...
 export const CLAIMS_REPO = 'https://raw.githubusercontent.com/gnosis/cow-mrkl-drop-data-chunks/final/chunks/'
 
-export enum ClaimType {
-  Airdrop, // free, no vesting, can be available on both mainnet and gchain
-  GnoOption, // paid, with vesting, must use GNO, can be available on both mainnet and gchain
-  UserOption, // paid, with vesting, must use Native currency, can be available on both mainnet and gchain
-  Investor, // paid, with vesting, must use USDC, only on mainnet
-  Team, // free, with vesting, only on mainnet
-  Advisor, // free, with vesting, only on mainnet
+export const enum ClaimType {
```
Comment (W3stside) [inline src/custom/state/claim/hooks/index.ts:25]: @alfetopito 's code i believe this was a numeric enum, 0,1,2,3,4 etc, is this intended to be string enum now?

Comment (alfetopito) [reply src/custom/state/claim/hooks/index.ts:25]: We discussed it today over the meeting but just for posterity sake, it should be a number enum as it was

**Comment by W3stside** [comment]
Superseded by #2054

**Pull request closed without merging**